src/gui/wizard/owncloudwizard.cpp

```diff
@@ -104,6 +108,55 @@ OwncloudWizard::OwncloudWizard(QWidget *parent)
 
     // allow Flow2 page to poll on window activation
     connect(this, &OwncloudWizard::onActivate, _flow2CredsPage, &Flow2AuthCredsPage::slotPollNow);
+
+    adjustWizardSize();
+}
+
+void OwncloudWizard::adjustWizardSize()
+{
+    const auto pageSizes = calculateWizardPageSizes();
+    const auto longestSide = calculateLongestSideOfWizardPages(pageSizes);
+
+    resize(QSize(longestSide, longestSide));
+}
+
+QList<QSize> OwncloudWizard::calculateWizardPageSizes() const
+{
+    QList<QSize> pageSizes;
+    const auto pIds = pageIds();
+
+    std::transform(pIds.begin(), pIds.end(), std::back_inserter(pageSizes), [this](int pageId) {
+        auto p = page(pageId);
+
+        p->adjustSize();
+
+        return p->sizeHint();
+    });
+
+    return pageSizes;
+}
+
+int OwncloudWizard::calculateLongestSideOfWizardPages(const QList<QSize> &pageSizes) const
+{
+    const auto calculateLongestSide = [](const QSize &size) {
+        return size.height() < size.width() ? size.width() : size.height();
+    };
+
+    return std::accumulate(
+        std::next(pageSizes.begin()),
+        pageSizes.end(),
+        [&pageSizes, &calculateLongestSide]() {
```

Comment:
@FlexW Why do we need this immediate lambda here? Can't we just go:
`
return std::accumulate(
        std::next(pageSizes.begin()),
        pageSizes.end(),
        calculateLongestSide(pageSizes.first()),
        [&calculateLongestSide](int currentLongestSide, const QSize &size) {
            const auto longestSideOfSize = calculateLongestSide(size);
            if (longestSideOfSize > currentLongestSide) {
                return longestSideOfSize;
            }
            return currentLongestSide;
        });
`
?

Reply:
This lambda makes indeed no sense there. I will remove it. It's a left over from developing this statement...